common/logging/src/sse_logging_components.rs

```diff
@@ -1,46 +1,110 @@
 //! This module provides an implementation of `slog::Drain` that optionally writes to a channel if
 //! there are subscribers to a HTTP SSE stream.
 
-use crate::async_record::AsyncRecord;
-use slog::{Drain, OwnedKVList, Record};
-use std::panic::AssertUnwindSafe;
+use once_cell::sync::Lazy;
+use serde_json;
+use serde_json::json;
+use serde_json::Value;
 use std::sync::Arc;
+use std::sync::Mutex;
 use tokio::sync::broadcast::Sender;
+use tracing::field::{Field, Visit};
+use tracing::{Event, Subscriber};
+use tracing_subscriber::layer::Context;
+use tracing_subscriber::Layer;
 
 /// Default log level for SSE Events.
 // NOTE: Made this a constant. Debug level seems to be pretty intense. Can make this
 // configurable later if needed.
-const LOG_LEVEL: slog::Level = slog::Level::Info;
-
+const LOG_LEVEL: tracing::Level = tracing::Level::INFO;
+pub static SSE_LOGGING_COMPONENTS: Lazy<Mutex<Option<SSELoggingComponents>>> =
+    Lazy::new(|| Mutex::new(None));
```

Comment:
My feeling is that this doesn't need to be a `static Lazy<Mutex>`.
The SSE logging only needs to be enabled in 2 locations, the BN HTTP API and the VC HTTP API both of which will be gated behind a CLI flag.

I think we should just be able to construct and use the layer only if those flags are present. But perhaps I'm misunderstanding the interaction between the SSE logging and the tracing subscriber which makes this mandatory? Let me know what you think.

Reply:
Don't remember the exact issue I was facing, but when I tried to go normally at first  I was getting blank logs when I was trying to curl,
I thought it had something to do with initialization & I needed to pass around the actual initialized instance of SSE Logging Components everywhere(in places like [this](https://github.com/sigp/lighthouse/blob/2e0eb6d1b8705bbda2ba56eb195d9cc7c6575e95/beacon_node/client/src/builder.rs#L815))(wasn’t sure if getting it from runtime context was working in my case) , so I went ahead with this approach and it ended up working

Not sure if what I did made too much sense though

Reply:
Ah okay I see. I'll have a play around with it and see if I can get something working :+1:

Reply:
I've added a commit with what I'm pretty sure is a working solution for SSE logging. Importantly, I've removed the `static`. This works for me locally, so let me know if it isn't working for you.

(I also ran `cargo fmt` so there is some formatting noise in that commit so just ignore that :sweat_smile:).

Reply:
Ahhh ,I think the issue I faced might have been caused because I removed the `sse_component_filter` part .
Thanks a lot !!